### ISSUE TITLE
[Feature Request] Filtering by glob

### ISSUE THREAD

**Issue by suessflorian** [kyazdani42/nvim-tree.lua#618]
Feature is present in https://github.com/ms-jpq/chadtree (GIF there showing it in action).

Suppose you have a typical project file tree like:
```
.
├── Dockerfile
├── README.md
├── api
│   ├── bulk
│   │   └── endpoint.go
│   └── graph
│       ├── endpoint.go
│       ├── generated
│       │   └── generated.go
│       ├── model
│       │   ├── models.go
│       │   ├── models_gen.go
│       │   └── namespaces
│       │       └── namespaces.go
│       ├── resolver
│       │   ├── common.resolvers.go
│       │   ├── filters.resolvers.go
│       │   ├── resolver.go
│       │   └── schema
│       │       ├── common.graphql
│       │       └── filters.graphql
│       └── signing
│           ├── signing.go
│           └── signing_test.go
├── config
│   └── config.go
├── config.yaml
├── db
│   └── migrations
│       └── 0000_average_spend.up.sql
├── docker-compose.yaml
├── go.mod
├── go.sum
├── gqlgen.yml
├── kube
│   └── dev.yaml
├── main.go
├── services
│   ├── ingestion.go
│   └── services.go
├── storage
│   ├── dataset.go
│   ├── filters.go
│   ├── filters_test.go
│   └── postgres.go
├── terraform
│   ├── dev-postgres.tf
│   └── main.tf
└── util
    ├── log
    │   └── log.go
    └── types
        └── decimal.go
```

and I want to see the tree of all the graphql files only via applying a filter `*.graphql`, I'd expect the tree to appear as

```
.
└── api
    └── graph
        └── resolver
            └── schema
                ├── common.graphql
                └── filters.graphql
```

Regardless if this is implemented or not, awesome project @kyazdani42, you're awesome!

**Comment by kyazdani42** [comment]
I would accept a PR i guess :)

**Comment by nadeemkhedr** [comment]
I was gonna write a new request that is very similar to this
Add a search that expands all folders containing the search term only, this will be super beneficial in mono repos especially

**Comment by mversic** [comment]
It would also be very useful if current filtering would accept glob patterns, not just literals, as in:
```
    filters = {
        dotfiles = false,
        custom = { '.*~', , '.*\.pyc$' }
    }
```

In the above example I don't ever want to see files ending in `~` or `pyc` files listed in directory tree

**Comment by alex-courtis** [comment]
A regex is like a glob...

**Comment by alex-courtis** [comment]
Also see #1056

**Comment by kyazdani42** [comment]
this was implemented in #1056
